instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/code/CodeAttributesGetter.java

```diff
@@ -17,8 +17,36 @@
 public interface CodeAttributesGetter<REQUEST> {
 
   @Nullable
-  Class<?> codeClass(REQUEST request);
+  default Class<?> getCodeClass(REQUEST request) {
+    return codeClass(request);
+  }
 
+  /**
+   * This method is deprecated and will be removed in the subsequent release.
+   *
+   * @deprecated Use {@link #getCodeClass(Object)} instead.
+   */
+  @Deprecated
   @Nullable
-  String methodName(REQUEST request);
+  default Class<?> codeClass(REQUEST request) {
+    throw new UnsupportedOperationException(
+        "This method is deprecated and will be removed in the subsequent release.");
```

Comment:
what do you think of using this pattern everywhere instead?

```suggestion
    return getCodeClass(request);
```

Reply:
Hmm, this would cause a recursive A->B->A->... situation; where if you forget to implement one of these methods you get a stack overflow error.